dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -613,14 +620,89 @@ export function JsonRpcContextProvider({
       }
     ),
   };
+  // -------- POLKADOT RPC METHODS --------
+
+  const polkadotRpc = {
+    testSignTransaction: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        // Below example is a scale encoded payload for system.remark("this is a test wallet-connect remark") transaction.
+        // decode url: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/extrinsics/decode/0x00019074686973206973206120746573742077616c6c65742d636f6e6e6563742072656d61726b
+        const transactionPayload =
+          "0x00019074686973206973206120746573742077616c6c65742d636f6e6e6563742072656d61726b05010000222400000d00000091b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3dc1f37ce7899cf20f63f5ea343f33e9e7b229c7e245049c2a7afc236861fc8b4";
+        try {
+          const result = await client!.request<{ payload: string; signature: string }>({
+            chainId,
+            topic: session!.topic,
+            request: {
+              method: DEFAULT_POLKADOT_METHODS.POLKADOT_SIGN_TRANSACTION,
+              params: {
+                address,
+                transactionPayload,
+              },
+            },
+          });
+          // ToDo: add sig validation after upgrating cra scripts (needs react-scripts ^v5.0.0)
+          /* 
+            await cryptoWaitReady();
+            const { isValid: valid } = signatureVerify(message, result.signature, address);
+          */
+          const valid = true;
+          return {
+            method: DEFAULT_POLKADOT_METHODS.POLKADOT_SIGN_TRANSACTION,
+            address,
+            valid,
+            result: result.signature,
+          };
+        } catch (error: any) {
+          throw new Error(error);
+        }
+      },
+    ),
+    testSignMessage: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        // some interfaces like sr25519 need to wait for WASM to load
```

Comment:
Is this leftover or referring to something specific in this method?

Reply:
The column indeed is misplaced, but it is related to the "await cryptoWaitReady()". In order to use sr25519 signatures in the lib it needs to wait till relevant WASM code loads. I moved the comment to the correct line.